### PR TITLE
[close #343]fix cdc residue metrics

### LINKED ISSUE #343: cdc: can't clean metrics when owner/processor quit unexpectedly
### 1. Describe the bug

1. startup a upstream cluster with cdc;
2. startup a downstream clsuter;
3. create a changefeed;
### 2. Minimal reproduce step (Required)

1. kill pd leader;

### 3. What did you see instead (Required)
1. No previous owner residual metrics;

### 4. What did you expect to see? (Required)
1. has

### 5. What is your migration tool and TiKV version? (Required)

- TiKV: v6.2.0
- TiKV CDC: v1.1.0

### PR DESCRIPTION
**Pull request by zeminzhou**
### What problem does this PR solve?

Issue Number: close #343

Problem Description: **TBD**

### What is changed and how does it work?
clean metrics againt

### Code changes

- Has exported function/method change
- Has exported variable/fields change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Integration test

### Side effects

- No side effects

### Related changes

- No related changes

### To reviewers

Please follow these principles to check this pull requests:

- **Concentration**. One pull request should only do one thing. No matter how small it is, the change does exactly one thing and gets it right. Don't mix other changes into it.
- **Tests**. A pull request should be test covered, whether the tests are unit tests, integration tests, or end-to-end tests. Tests should be sufficient, correct and don't slow down the CI pipeline largely.
- **Functionality**. The pull request should implement what the author intends to do and fit well in the existing code base, resolve a real problem for TiDB/TiKV users. To get the author's intention and the pull request design, you could follow the discussions in the corresponding GitHub issue or [internal.tidb.io](https://internals.tidb.io) topic.
- **Style**. Code in the pull request should follow common programming style. *(For Go, we have style checkers in CI)*. However, sometimes the existing code is inconsistent with the style guide, you should maintain consistency with the existing code or file a new issue to fix the existing code style first.
- **Documentation**. If a pull request changes how users build, test, interact with, or release code, you must check whether it also updates the related documentation such as READMEs and any generated reference docs. Similarly, if a pull request deletes or deprecates code, you must check whether or not the corresponding documentation should also be deleted.
- **Performance**. If you find the pull request may affect performance, you could ask the author to provide a benchmark result.

*(The above text mainly refers to [TiDB Development Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/review-a-pr.html#checking-pull-requests). It's also highly recommended to read about [Writing code review comments](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/review-a-pr.html#writing-code-review-comments))*

### CODE REVIEW

File: cdc/cdc/owner/owner.go [hunk 1]
```diff
@@ -199,6 +199,14 @@ func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState)
 	return state, nil
 }
 
+// Close try to close the owner when the owner is not running.
+// Note: This method must not be called with `Tick`. Please be careful.
+func (o *Owner) Close(ctx cdcContext.Context) {
```
Comment (pingyu) [inline cdc/cdc/owner/owner.go:204]: Suggest to name as `CloseAllCaptures`, as we are not closing `Owner` here.

[Here](https://github.com/tikv/migration/blob/73d43788dfc2513d5bc3e579363438f147ccfb7d/cdc/cdc/owner/owner.go#L191) has similar code block, suggest to reuse this function.

(And maybe we also need the wrapper of context.)

At last, we need to double check about the thread-safety.

File: cdc/cdc/processor/manager.go [hunk 2]
```diff
@@ -155,6 +155,14 @@ func (m *Manager) sendCommand(tp commandTp, payload interface{}) chan struct{} {
 	return cmd.done
 }
 
+// Close closes all processors.
+// Note: This method must not be called with `Tick`. Please be careful.
+func (m *Manager) Close() {
```
Comment (pingyu) [inline cdc/cdc/processor/manager.go:160]: Suggest to name as `SyncClose`, and reuse in handling of `commandTpClose` command.

**Review by haojinming — APPROVED** [review]
LGTM

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**